src/utils.jl

```diff
@@ -104,20 +104,26 @@ function to_vectorSvector(a::AbstractMatrix)
 end
 
 """
-    orthonormal(D, k) -> ws
+    orthonormal([T,] D, k) -> ws
 Return a matrix `ws` with `k` columns, each being
 an `D`-dimensional orthonormal vector.
 
-Always return `SMatrix` for stability reasons.
+Default: returns SMatrix{D, k} if D*k < 100, otherwise Matrix
 """
 function orthonormal end
 
-@inline function orthonormal(D::Int, k::Int)
+orthonormal(D, k) = D*k < 100 ? orthonormal(SMatrix, D, k) : orthonormal(Matrix, D, k)
```

Comment:
This is a breaking change. But to be honest, this is definitely worth doing... But please mention this in the documentation string. What you say now is not true, thje default is not to return `SMatrix`, because the default depends on the input.

Reply:
So, breaking change meaning increasing the major version? I'm confused..